pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,305 @@
+        google-cloud-storage
+        ipython
+        jinja2
+        json5
+        jsonschema
+        lark
+        mobly
+        mypy
+        mypy-extensions
+        mypy-protobuf
+        packaging
+        parameterized
+        pip-tools
+        prompt-toolkit
+        protobuf
+        psutil
+        ptpython
+        pyelftools
+        pygments
+        pykwalify
+        pylint
+        pyperclip
+        pyserial
+        python-daemon
+        pyyaml
+        requests
+        setuptools
+        six
+        sphinx
+        sphinx-argparse
+        sphinx-design
+        stringcase
+        toml
+        tornado
+        types-protobuf
+        types-pyyaml
+        types-requests
+        types-setuptools
+        watchdog
+        websockets
+        wheel
+        yapf
+      ];
+      depListToAttrs =
+        list:
+        builtins.listToAttrs (
+          map (dep: {
+            name = dep.name;
+            value = dep;
+          }) list
+        );
+      saturateDependencies =
+        deps:
+        let
+          before = deps;
+          new = lib.mergeAttrsList (
+            map (dep: depListToAttrs (dep.propagatedBuildInputs or [ ])) (lib.attrValues before)
+          );
+          after = before // new;
+        in
+        if lib.attrNames before != lib.attrNames after then saturateDependencies after else before;
+      saturateDependencyList = list: lib.attrValues (saturateDependencies (depListToAttrs list));
+      saturatedDependencyList = lib.filter (drv: drv ? dist) (saturateDependencyList dependencies);
+    in
+    lib.concatMapStringsSep " " (dep: "file://${dep.dist}") saturatedDependencyList;
+
+  configurePhase =
+    let
+      configureArguments = [
+        ''chip_project_config_include_dirs=["//.."]''
+        ''chip_crypto="boringssl"''
+        ''enable_rtti=true''
+        ''chip_config_memory_debug_checks=false''
+        ''chip_config_memory_debug_dmalloc=false''
+        ''chip_mdns="minimal"''
+        ''chip_minmdns_default_policy="libnl"''
+        ''chip_python_version="${lib.versions.majorMinor python3.version}"''
+        ''chip_python_platform_tag="any"''
+        ''chip_python_package_prefix="home-assistant-chip"''
+        ''custom_toolchain="custom"''
+        ''target_cc="${stdenv.cc.targetPrefix}cc"''
+        ''target_cxx="${stdenv.cc.targetPrefix}c++"''
+        ''target_ar="${stdenv.cc.targetPrefix}ar"''
+      ];
+    in
+    ''
+      runHook preConfigure
+
+      gn --root=. gen --check --fail-on-unused-args out ${
+        lib.escapeShellArg ("--args=" + (lib.concatStringsSep " " configureArguments))
+      }
+
+      runHook postConfigure
+    '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    export NIX_CFLAGS_COMPILE="$($PKG_CONFIG --cflags glib-2.0) -O2 -Wno-error"
+    export NIX_CFLAGS_LINK="$($PKG_CONFIG --libs gio-2.0) $($PKG_CONFIG --libs gobject-2.0) $($PKG_CONFIG --libs glib-2.0)"
+
+    ninja -C ./out -j$NIX_BUILD_CORES chip-repl
```

Comment:
tiny optional nit: Similarly to the gn hooks, nixpkgs also has `ninja` hooks that I think would work nicely. You can set `chip-repl` in `ninjaFlags`
https://github.com/NixOS/nixpkgs/blob/6ced9c6711a42e6b1195384709c4af2f79ea7893/pkgs/by-name/ni/ninja/setup-hook.sh#L19